### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by yarom82**

### CODE REVIEW

File: src/initial-state.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  leftPlayerDown: true,
+  rightPlayerDown: true
+  /* game: {
+    leftPlayer: {
+      down = true
+    },
+    rightPlayer: {
+      down = true
+    }
+  } */
```
Comment (mightyiam) [inline src/initial-state.js]: Correct state names:
* `leftHiding`
* `rightHiding`

Correct action names:
* `LEFT_HIDE`
* `RIGHT_HIDE`
* `LEFT_UNHIDE`
* `RIGHT_UNHIDE`

Checklist:
* [ ] Action names
* [ ] State key names
* [ ] Stream names

File: src/make-ui.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  return span(
+    {style: {
+      display: 'inline-block',
+      transform: side === 'right' ? 'scale(-1,1)' : null
+    }},
+    hiding ? 'd' : 'D'
+  )
+}
+
+const curriedMkPlayer = R.curry(player)
```
Comment (mightyiam) [inline src/make-ui.js]: `curriedPlayer`

File: src/make-ui.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const leftPlayer = curriedMkPlayer('left')
+const rightPlayer = curriedMkPlayer('right')
+const barrier = '='
+
+const makeUI = (state) => {
```
Comment (mightyiam) [inline src/make-ui.js]: No need for parens.

File: src/state-machine.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// todo: immutablejs
```
Comment (mightyiam) [inline src/state-machine.js]: I don't like these. I prefer this in the issues. Regarding this specific one, we'll know when we require it.

File: src/state-machine.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      break
+    case 'RIGHT_DOWN':
+      newState.rightPlayerDown = true
+      break
+    case 'RIGHT_UP':
+      newState.rightPlayerDown = false
+      break
+  }
+  return newState
+}
```
Comment (mightyiam) [inline src/state-machine.js]: Great!

File: src/state-machine.spec.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    {
+      currentState: { rightPlayerDown: true },
+      expectedState: { rightPlayerDown: false }
+    }
+  ]
+}
+
+for (const action in actionTests) {
+  actionTests[action].forEach((test, index) => {
```
Comment (mightyiam) [inline src/state-machine.spec.js]: For readability, make `actionTests[action]` a constant. Such as `const tests = actionTests[action]`.

**Comment by yarom82** [comment]
* [ ] package.json: cyclejs-shooter in 
* [ ] package.json: Cycle.js-based hide-and-shoot game
* [ ] package.json: typo
* [ ] bootsrap.js: shooter should be extracted into a constant.
* [ ] index.js: merge should be a one-liner
* [ ] index.js: Do not leave debugging in production.
* [ ] make-ui.js: Better name for this is uiFromState
* [ ] Correct state names
* [ ] Correct action names
* [ ] index.js: stream names
* [ ] make-ui.js: curriedPlayer
* [ ] state-machine.js: remove comment
* [ ] spec: For readability, make actionTests[action] a constant. Such as const tests = actionTests[action].

**Pull request merged**